uECC.c

```diff
@@ -1233,6 +1233,7 @@ static int uECC_sign_with_k(const uint8_t *private_key,
                             const uint8_t *message_hash,
                             unsigned hash_size,
                             uECC_word_t *k,
+                            uint8_t *recid,
```

Comment:
Instead of adding another parameter, you could return the recid in the first byte of k. That would reduce the overhead when people don't care about the recid.